Harvesting data from OJS

Kirubamoh opened the pull request.

Kirubamoh committed:
Add html parser to requirments

Kirubamoh committed:
Merge branch 'ifgi:main' into main

Kirubamoh committed:
Add test files

Kirubamoh committed:
Add models for OJSservers

Kirubamoh committed:
Add harvestinh=g functions to tasks.py #11

Kirubamoh committed:
Clean unused views and Urls

Kirubamoh committed:
add links of article to popup

nuest commented on publications/tasks.py:
```diff
@@ -1,22 +1,71 @@
+            type_geom.update(geom_content)
+            geom_data = json.dumps(type_geom)
+            # geometry field accept json object as string
+            geom_object = GEOSGeometry(geom_data) #GeometryCollection object
```
This could be something that breaks in the real world, so should handle such edge cases here.

- add test case with invalid geometry
- add test case with invalid JSON

nuest commented on publications/tasks.py:
```diff
@@ -1,22 +1,71 @@
+    articles = collection.getElementsByTagName("dc:identifier")                                            
+    articles_count_in_journal = len(articles)  # number of articles in journal
+    for i in range(articles_count_in_journal):
+        identifier = collection.getElementsByTagName("dc:identifier")
```
Extract the parsing of a single article into an extra function please.

nuest commented on tests/test_htmlparser.py:
```diff
@@ -0,0 +1,32 @@
+        self.json_object = parse_html(self.soup)
+        self.assertEqual(self.json_object['type'], 'FeatureCollection')
+        self.assertEqual(self.json_object["features"][0]["geometry"]["type"], 'LineString')
+        self.assertIsNotNone(self.json_object)
```
Add a test that parses a full article (take from the test files) and then

- saves the publication in the database
- check that the API now returns the publication and the information is correct

nuest replied:
Done in https://github.com/ifgi/optimetaPortal/pull/76/files#diff-34c98f76d2c694bcb7939bb20927517dc8a9745aeb9b7d855d7e1c9465ba54f0

nuest commented on tests/test_xml_idenitifier.py:
```diff
@@ -0,0 +1,63 @@
+
+        httpretty.enable(verbose=True, allow_net_connect=False)  # enable HTTPretty so that it will monkey patch the socket module
+        httpretty.register_uri(httpretty.GET, "http://localhost:8330/index.php/opti-geo/article/view/1",
+                            body="""
```
You could also load the content from the HTML files, `body = readFile(..)`.

https://www.askpython.com/python/examples/read-file-as-string-in-python

Kirubamoh committed:
modify values into popup

Kirubamoh committed:
modify values into popup

Kirubamoh committed:
Add subscription model and update fields

Kirubamoh committed:
Update Harvest functions

Kirubamoh committed:
Update Models

Kirubamoh committed:
Update test data

Kirubamoh committed:
Update popup with timeperiod values

Kirubamoh commented:
#34  updated popup

Kirubamoh committed:
Add serialisers to subscription Model #39

Kirubamoh committed:
Add viewsets and api path to subscription Model #39

Kirubamoh committed:
Update subscription Model #39

Kirubamoh committed:
Implement User subscription page #39

Kirubamoh committed:
Implement usersubscription page

Kirubamoh committed:
Add new tests and update test

Kirubamoh committed:
Add new tests and update test

Kirubamoh committed:
Add new tests and update test

Kirubamoh committed:
update subscriptions.js

nuest commented on tests/test_xml_idenitifier.py:
```diff
@@ -0,0 +1,63 @@
+        return identifier_value
+
+
+    def test_one(self):
```
Give the test a more informative name please.

Kirubamoh committed:
update test name

Kirubamoh commented on publications/views.py:
```diff
@@ -113,7 +76,24 @@ def user_settings(request):
+        start_date = request.POST.get('start_date', False)
+        end_date = request.POST.get('end_date', False)
+        currentuser = request.user
+        user_name = currentuser.username
```
Test case is not added. Over to you @nuest

Kirubamoh committed:
Update urls.py

Clear indentations

Kirubamoh committed:
Update subscriptions.js

remove  unneccesary symbols

Kirubamoh committed:
update user subscription

Kirubamoh committed:
def user_subscriptions(request):
    if request.user.is_authenticated:
        subs = Subscription.objects.all()
        count_subs = Subscription.objects.all().count()
        return render(request,'subscriptions.html',{'sub':subs,'count':count_subs})
    else:
        pass
    erge branch 'main' of https://github.com/Kirubamoh/optimetaPortal into main

Kirubamoh commented on publications/templates/subscriptions.html:
```diff
@@ -1,2 +1,109 @@
+{% endblock %}
+
+{% block scripts %}
+<script src="https://cdnjs.cloudflare.com/ajax/libs/bootstrap-datepicker/1.9.0/js/bootstrap-datepicker.min.js" integrity="sha512-T/tUfKSV1bihCnd+MxKD0Hm1uBBroVYBOYSk1knyvQ9VyZJpc/ALb4P0r6ubwVPSGB2GvjeoMAJJImBG12TiaQ==" crossorigin="anonymous" referrerpolicy="no-referrer"></script>
```
https://github.com/uxsolutions/bootstrap-datepicker 
https://cdnjs.com/libraries/bootstrap-datepicker

Please add JS files @nuest

Kirubamoh committed:
add timeline to main page

Kirubamoh commented:
Please add relative JS packages for date picker and timeline. Currently they are accessed from cdn. @nuest

Kirubamoh committed:
Update .gitignore
